apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/CustomIdGeneratorTests.kt

```diff
@@ -0,0 +1,74 @@
+package com.apollographql.apollo.gradle.test
+
+import com.apollographql.apollo.gradle.util.TestUtils
+import com.apollographql.apollo.gradle.util.TestUtils.withSimpleProject
+import com.apollographql.apollo.gradle.util.replaceInText
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Assert
+import org.junit.Test
+import java.io.File
+
+class CustomIdGeneratorTests {
+  @Test
+  fun `up-to-date checks are working`() {
+    val apolloConfiguration = """
+      class MyIdGenerator implements CustomIdGenerator {
+          String apply(String queryString, String queryFilepath) {
+              return queryString.length().toString();
+          }
+          String version = "1"
+      }
+      
+      apollo {
+        customIdGenerator = new MyIdGenerator()
+      }
+    """.trimIndent()
+
+    withSimpleProject(apolloConfiguration = apolloConfiguration) {dir ->
+      val gradleFile = File(dir, "build.gradle").readText()
+
+      File(dir, "build.gradle").writeText("import com.apollographql.apollo.compiler.CustomIdGenerator\n$gradleFile")
+
+      var result = TestUtils.executeTask("generateApolloSources", dir)
+
+      Assert.assertEquals(TaskOutcome.SUCCESS, result.task(":generateApolloSources")!!.outcome)
+
+      result = TestUtils.executeTask("generateApolloSources", dir)
+
+      Assert.assertEquals(TaskOutcome.UP_TO_DATE, result.task(":generateApolloSources")!!.outcome)
+    }
+  }
+
+  @Test
+  fun `changing the customIdGenerator recompiles sources`() {
+    val apolloConfiguration = """
+      class MyIdGenerator implements CustomIdGenerator {
+          String apply(String queryString, String queryFilepath) {
+              return queryString.length().toString();
+          }
+          String version = "1.0.0-SNAPSHOT"
```

Comment:
Nitpicking: What about `CustomIdGenerator-v1` or something similar here ? `-SNAPSHOT` feels very maven-oriented and isn't very relevant here.

Reply:
I just used it to be unique to search and replace below. I'll change it to the `MyIdGenerator-v1`